src/connector/src/sink/iceberg/mod.rs

```diff
@@ -320,7 +375,82 @@ impl IcebergConfig {
         Ok(iceberg_configs)
     }
 
+    fn build_jni_catalog_configs_for_nimtable(
+        &self,
+    ) -> Result<(BaseCatalogConfig, HashMap<String, String>)> {
+        let mut iceberg_configs = HashMap::new();
+
+        let base_catalog_config = {
+            let catalog_type = self.catalog_type().to_string();
+
+            iceberg_configs.insert(CATALOG_TYPE.to_string(), catalog_type.clone());
+            iceberg_configs.insert(CATALOG_NAME.to_string(), self.catalog_name());
+
+            if let Some(region) = &self.region {
+                // icelake
+                iceberg_configs.insert(
+                    "iceberg.table.io.region".to_string(),
+                    region.clone().to_string(),
+                );
+                // iceberg-rust
+                iceberg_configs.insert(
+                    ("iceberg.table.io.".to_string() + S3_REGION).to_string(),
+                    region.clone().to_string(),
+                );
+            }
+
+            let (bucket, _) = {
+                let url = Url::parse(&self.path).map_err(|e| SinkError::Iceberg(anyhow!(e)))?;
+                let bucket = url
+                    .host_str()
+                    .ok_or_else(|| {
+                        SinkError::Iceberg(anyhow!(
+                            "Invalid s3 path: {}, bucket is missing",
+                            self.path
+                        ))
+                    })?
+                    .to_string();
+                let root = url.path().trim_start_matches('/').to_string();
+                (bucket, root)
+            };
+
+            iceberg_configs.insert("iceberg.table.io.bucket".to_string(), bucket);
```

Comment:
Observe L395, specialise the region path of `icelake` and `iceberg-rust`. May I ask if the bucket path needs the same treatment? Can you provide the link in the doc?

Reply:
The reason why bucket is needed is both icelake and iceberg-rust use opendal under the hood, so bucket is needed. However, the region is handled differently by icelake and iceberg-rust. They use different name to parse the config map.